lib/tasks/whitehall_importer.rb

```diff
@@ -169,7 +169,11 @@ def set_withdrawn_status(whitehall_edition, edition)
         revision_at_creation: edition.revision,
         created_by_id: user_ids[event["whodunnit"]],
         created_at: event["created_at"],
-        details: state_details(whitehall_edition, edition),
+        details: Withdrawal.new(
```

Comment:
In @1pretz1's [PR](https://github.com/alphagov/content-publisher/pull/1439/commits/2badfcc1c6a2425259a2bc52e0236cdeb03d3ba3#diff-713c586411d3e8a35468b374fe29c4a8R205) a `Removal` is added to details, so we may want to leave this as a separate method.

Reply:
I don't think we should look to making that a big method that handles all of those bits, it's only going to get complex.